source/install/verify-signatures/macos.txt

```diff
@@ -0,0 +1,142 @@
+.. _verify-signatures-macos:
+
+======================================
+Verify MongoDB Shell Packages on macOS
+======================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: tutorial
+
+.. include:: /includes/verify-signatures-intro.rst
+
+Before you Begin
+----------------
+
+If you don't have MongoDB Shell installed, download the MongoDB Shell
+binary from the `Download Center
+<https://www.mongodb.com/try/download/shell?jmp=docs>`__.
+
+Steps
+-----
+
+To verify MongoDB Shell packages on macOS, you can use one of these
+methods:
+
+- :ref:`Disk Image Verification <verify-signatures-macos-disk-image>`
+
+- :ref:`GPG <verify-signatures-macos-gpg>`
+
+.. _verify-signatures-macos-disk-image:
+
+Verify Packages with Disk Image Verification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following command:
+
+.. code-block:: sh
+
+   codesign -dv --verbose=4 <path_to_mongosh>
+
+If the package is properly signed by MongoDB, the output includes the
+following information:
+
+.. code-block:: sh
+   :copyable: false
+
+   Authority=Developer ID Application: MongoDB, Inc. (4XWMY46275)
+   Authority=Developer ID Certification Authority
+   Authority=Apple Root CA
+
+.. _verify-signatures-macos-gpg:
+
+Verify Packages with GPG
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. procedure::
+   :style: normal
+
+   .. step:: Import the MongoDB Shell public key
+
+      .. code-block:: sh
+
+         curl https://pgp.mongodb.com/mongosh.asc | gpg --import
+
+      If the key imports successfully, the command returns the
+      following output:
+
+      .. code-block:: sh
+         :copyable: false
+
+         gpg: key CEED0419D361CB16: public key "Mongosh Release Signing Key <packaging@mongodb.com>" imported
+         gpg: Total number processed: 1
+         gpg:               imported: 1
+
+      If you have previously imported the key, the command returns the
+      following output:
+
+      .. code-block:: sh
+         :copyable: false
+
+         gpg: key A8130EC3F9F5F923: "Mongosh Release Signing Key <packaging@mongodb.com>" not changed
+         gpg: Total number processed: 1
+         gpg:             	      unchanged: 1
+
+   .. step:: Download the MongoDB Shell public signature
+
+      To download the MongoDB Shell public signature, go to the `mongosh
+      Releases <https://github.com/mongodb-js/mongosh/releases>`__ page
+      on GitHub and download the corresponding ``.sig`` file for your
+      version and variant.
+
+      For example, if you are running
+      ``mongosh-{+version+}-linux-arm64``, download
+      ``mongosh-{+version+}-linux-arm64.sig``
```

Comment:
That's a good point, it could maybe be generalized to apply to all OSes... I was basing these instructions on this [doc](https://docs.google.com/document/d/1ovw8S4nKhA9ZpncHNpDQTSeziYwLIfDOPHd9s-yGAqM/edit), which separates instructions by OS. (I'm not sure who the original author of that doc is...)

My understanding from that doc is that the GPG approach can be used for all OSes, so do you think we could/should only document the GPG approach on this page and not make it OS-specific?

Reply:
We don't have GPG signatures for macOS, though, so that part _is_ macOS-specific

@mabaasit seems to be the author of that doc, so pinging him here

Reply:
@jeff-allen-mongo as @addaleax pointed out, we do not have GPG signed artifacts for mongosh, so this section is not really required for mac, but can be a general one. The doc you shared was written specifically for Compass (where we have GPG signed artifacts for macos).